src/middleware/verifySingup.ts

```diff
@@ -0,0 +1,47 @@
+// import { ROLES } from "../models/Role"
+
+import { Request, Response, NextFunction } from 'express';
+import { UserModel } from "../routes/schemas/User";
+import mongoose from 'mongoose';
+
+export const checkDuplicateUsernameOrEmail = async (req: Request, res: Response, next: NextFunction) => {
+  const User = UserModel(mongoose); 
+
+  const email = await User.findOne({ email: req.body.email });
+
+  if (email) {
+    res.status(400).json({ message: 'The email already exists' });
```

Comment:
please install the library npm install http-status-codes and manage from there the status codes